pkg/aspect/root/flags/interceptor.go

```diff
@@ -86,32 +86,68 @@ func FlagsInterceptor(streams ioutils.Streams) interceptors.Interceptor {
 		if cfgFile != "" {
 			// Use config file from the flag.
 			viper.SetConfigFile(cfgFile)
+			viper.AutomaticEnv()
+			if err := viper.ReadInConfig(); err == nil {
+				faint.Fprintln(streams.Stderr, "Using config file:", viper.ConfigFileUsed())
+			}
 		} else {
-			// Find home directory.
-			home, err := homedir.Dir()
-			cobra.CheckErr(err)
-
-			// Search for config in home directory with name ".aspect" (without extension).
-			viper.AddConfigPath(home)
+			// used to search in current directory
+			viper.AddConfigPath(".")
 			viper.SetConfigName(".aspect")
 
-			// Search for config in root of current repo with name ".aspect" (without extension).
-			repoViper.AddConfigPath(".")
-			repoViper.SetConfigName(".aspect")
-			repoViper.AutomaticEnv()
-		}
-
-		viper.AutomaticEnv()
-		if err := viper.ReadInConfig(); err == nil {
-			faint.Fprintln(streams.Stderr, "Using config file:", viper.ConfigFileUsed())
-		}
-
-		if err := repoViper.ReadInConfig(); err == nil {
-			faint.Fprintln(streams.Stderr, "Using config file:", repoViper.ConfigFileUsed())
+			// used to search in home directory
+			home, err := homedir.Dir()
+			cobra.CheckErr(err)
+	
+			viper.AutomaticEnv()
+
+			// what do we want to do with error handeling for all of these reads?
+			// it should be consistent just not sure what it should be
+			// the errors that aren't viper.ConfigFileNotFoundError will still be thrown how should they be handled?
+
+			// search in current directory first
+			if err := viper.ReadInConfig(); err != nil {
+				if _, ok := err.(viper.ConfigFileNotFoundError); ok {
+					// search in home directory next
+					viper.AddConfigPath(home)
+					if err := viper.ReadInConfig(); err != nil {
+						if _, ok := err.(viper.ConfigFileNotFoundError); ok {
+							// Config file not found;
+							// create new config file if it does not exist
+							if err := viper.WriteConfigAs(fmt.Sprintf("%s/.aspect.yaml", home)); err != nil {
+								// if you can't write to home directory fail and write to the current directory
+								// Should we handle if this fails?
+								viper.WriteConfigAs(".aspect.yaml")
```

Comment:
Write relative to the working directory? I think you actually want to pass the absolute path here by using the workspace path.

Reply:
I was trying to line up the earlier viper.AddConfigPath(".") and this write so that they would be in the same place. If there is a different way we would like to resolve the the workspace path to an absolute path we should initialize that earlier so the read and write looking in the same place, no?